### PR TITLE
NOOS-432/update-boilerplate

### PR DESCRIPTION
**Pull request by filiperosa**
JIRA Reference: [NOOS-432](https://noosenergy.atlassian.net/browse/NOOS-423)

[NOOS-432]: https://noosenergy.atlassian.net/browse/NOOS-432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,7 +18,7 @@ pip-wheel-metadata
 
 # PyTest
 .pytest_cache
-pytest.xml
+.pytest_results
 geckodriver.log
 
 # MyPy
```
Comment (pieeri) [inline .gitignore:24]: Would it be possible to add a filter as well for 

```
# Pyenv
.python-version
```

File: pyproject.toml [hunk 2]
```diff
@@ -91,5 +93,9 @@ python_files = "test_*.py"
 python_classes = "Test"
 python_functions = "test_*"
 junit_family = "xunit2"
-filterwarnings = "error::RuntimeWarning"
-addopts = "--headless --webdriver=Firefox --junitxml=pytest.xml"
+filterwarnings = [
+    "error::RuntimeWarning",
+    "ignore::DeprecationWarning",
```
Comment (pieeri) [inline pyproject.toml:98]: ❓ 

Are still these filters required?

**Review by pieeri — CHANGES_REQUESTED** [review]
Very good job to resume the CI/CD pipeline on such a library!

To tidy up the breadcrumbs, few suggestions included. In addition, would it be possible to:

- remove the deprecated variable `PYTHONBREAKPOINT=ipdb.set_trace` in the project `.env` file template?
- update the CircleCI token for displaying the pipeline status in the project `README`?

File: .gitignore (hunk 1 above)
Comment (filiperosa) [reply .gitignore:24]: Done 👍

File: pyproject.toml [hunk 3]
```diff
@@ -45,10 +45,11 @@ matplotlib = "*"
...
 # https://github.com/plotly/dash/issues/1992
-werkzeug = ">=2.0,<2.1"
+werkzeug = "2.0.1"
 
 [tool.poetry.dev-dependencies]
 dash = { version = ">=1.15.0,<2.0.0", extras = ["dev", "testing"] }
+selenium = "<=4.2.0" # Upper versions require Dash v2.**
 # Format
 isort = "*"
 black = {version = "*", allow-prereleases = true}
```
[the remark this answers is not in the document]
Comment (filiperosa) [reply pyproject.toml:54]: Done 👍

File: pyproject.toml [hunk 4]
```diff
@@ -91,5 +91,9 @@ python_files = "test_*.py"
...
 python_functions = "test_*"
 junit_family = "xunit2"
-filterwarnings = "error::RuntimeWarning"
-addopts = "--headless --webdriver=Firefox --junitxml=pytest.xml"
+filterwarnings = [
+    "error::RuntimeWarning",
+    "ignore::DeprecationWarning",
+    "ignore::UserWarning"
+]
+addopts = "--headless --webdriver=Firefox --junitxml=pytest.xml"
```
[the remark this answers is not in the document]
Comment (filiperosa) [reply pyproject.toml]: Good idea! Added `"files.insertFinalNewline": true` to the the vscode settings

File: .circleci/config.yml [hunk 5]
```diff
@@ -91,43 +103,43 @@ workflows:
...
-            - test-package
+            - lint_package
+            - test_package
 
   publish:
     jobs:
-      - approve-for-publishing:
+      - approve_for_publishing:
           type: approval
           filters:
```
[the remark this answers is not in the document]
Comment (filiperosa) [reply .circleci/config.yml]: Solved 👍

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+    "editor.formatOnSave": true,
+    "files.insertFinalNewline": true,
```
Comment (pieeri) [inline .vscode/settings.json:3]: 🐼 

Not required ; such a formatting rule should be already controlled by the cross-IDE/text editor `.editorconfig` file!

https://editorconfig.org/
https://github.com/noosenergy/dash-mdc-neptune/blob/master/.editorconfig#L10

**Review by pieeri — CHANGES_REQUESTED** [review]
Last suggestion ; thanks a lot for such a boilerplate revamp!

File: .vscode/settings.json (hunk 6 above)
Comment (filiperosa) [reply .vscode/settings.json]: Thanks, removed it!

**Review by pieeri — APPROVED** [review]
Let's ship!

**Review by pieeri — APPROVED** [review]
🚢

**Pull request merged**